azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -0,0 +1,499 @@
+		}
+	}
+
+	if props := existing.Properties; props != nil {
+		return tf.ImportAsExistsError("azurerm_subscription", id.ID())
+	}
+
+	workload := subscriptionAlias.Production
+	if workloadRaw := d.Get("workload").(string); workloadRaw != "" {
+		workload = subscriptionAlias.Workload(workloadRaw)
+	}
+
+	req := subscriptionAlias.PutAliasRequest{
+		Properties: &subscriptionAlias.PutAliasRequestProperties{
+			Workload: workload,
+		},
+	}
+
+	targetState := d.Get("state").(string)
+	reactivate := false
+	subscriptionId := ""
+
+	// Check if we're adding alias management for an existing subscription
+	if subscriptionIdRaw, ok := d.GetOk("subscription_id"); ok {
+		subscriptionId = subscriptionIdRaw.(string)
+		exists, err := checkExistingAliases(ctx, *aliasClient, subscriptionId)
+		if err != nil {
+			return err
+		}
+		if exists != nil {
+			return fmt.Errorf("An Alias for Subscription %q already exists with name %q - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for %q for more information.", subscriptionId, *exists, "azurerm_subscription")
+		}
+
+		req.Properties.SubscriptionID = utils.String(subscriptionId)
+		existingSub, err := client.Get(ctx, subscriptionId)
+		if err != nil {
+			return fmt.Errorf("could not read existing Subscription %q", subscriptionId)
+		}
+		// Disabled and Warned are both "effectively" cancelled states,
+		if (existingSub.State == subscriptions.Disabled || existingSub.State == subscriptions.Warned) && targetState == "Active" {
+			log.Printf("[DEBUG] Existing subscription already in use in Disabled state Terraform will attempt to re-activate it")
+			reactivate = true
+		}
+	} else {
+		// If we're not assuming control of an existing Subscription, we need to know where to create it.
+		req.Properties.DisplayName = utils.String(d.Get("subscription_name").(string))
+
+		billingAccount := d.Get("billing_account").(string)
+
+		if enrollmentAccount, ok := d.GetOk("enrollment_account"); ok && enrollmentAccount.(string) != "" {
+			req.Properties.BillingScope = utils.String(fmt.Sprintf(billingScopeEnrollmentFmt, billingAccount, enrollmentAccount))
+		} else {
+			billingProfile := d.Get("billing_profile").(string)
+			invoiceSection := d.Get("invoice_section").(string)
+			req.Properties.BillingScope = utils.String(fmt.Sprintf(billingScopeMCAFmt, billingAccount, billingProfile, invoiceSection))
+		}
+	}
+
+	future, err := aliasClient.Create(ctx, aliasName, req)
+	if err != nil {
+		return fmt.Errorf("creating new Subscription (Alias %q): %+v", aliasName, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, aliasClient.Client); err != nil {
+		return fmt.Errorf("waiting for creation of Subscription with Alias %q: %+v", id.Name, err)
+	}
+
+	if reactivate {
+		if _, err := subscriptionClient.Enable(ctx, subscriptionId); err != nil {
+			return fmt.Errorf("enabling Subscription %q: %+v", subscriptionId, err)
+		}
```

Comment:
do we need to wait for this to settle first?

Reply:
yes, and this has also moved as a result of removing the `state` property noted above.